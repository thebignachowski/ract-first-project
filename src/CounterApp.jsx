import { useState } from 'react';
import PropsTypes from 'prop-types';


export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value )

    const handleClick = () => setCounter( (c) => c + 1 )
    const handleSubstract = () => setCounter( (c) => c - 1 )
    const handleReset = () => setCounter( value )

    return (
        <>
            <h1>Counter App</h1>
            <h2> { counter } </h2>
            <button onClick={handleClick}>+1</button>
            <button onClick={handleSubstract}>-1</button>
            <button onClick={handleReset}>Reset</button>
        </>
    )
    
}

CounterApp.PropsTypes = {
    value: PropsTypes.number.isRequired
}