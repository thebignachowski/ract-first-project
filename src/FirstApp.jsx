export const App = () => {
    const name = 'Nacho Rodríguez'
    return <h1>Hola { name }, qué tal tú!!!</h1>
}