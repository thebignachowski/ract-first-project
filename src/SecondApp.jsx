import PropTypes from 'prop-types';

export const SecondApp = ( { title, subtitle, name }) => {
    
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    )
}

SecondApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

SecondApp.defaultProps = {
    name: 'Nachowski',
    subtitle: 'No hay subtítulo',
    //title: 'No hay ningún título',
}