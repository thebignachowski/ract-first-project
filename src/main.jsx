import React from 'react';
import ReactDOM from 'react-dom/client';

import { CounterApp } from './CounterApp';
import { SecondApp } from './SecondApp';
import './css/style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp  value={ 0 } />
        { /*<SecondApp title="Hola, Soy Vegeta"/> */}
    </React.StrictMode>
)

