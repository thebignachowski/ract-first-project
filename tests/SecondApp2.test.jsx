import { render, screen } from "@testing-library/react";
import { SecondApp } from "../src/SecondApp";


describe('Pruebas en <SecondApp />', () => { 

    const title = 'Hola, Soy Goku';
    const subtitle = 'Soy un subtítulo';
    test('debe hacer match con el snapshot', () =>{        
        const { container } = render( <SecondApp title={ title } />);    
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola, Soy Goku"', () => {
        render( <SecondApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe de mostrar el título en un h1', () =>{
        render( <SecondApp title={title} />);
        expect( screen.getByRole('heading', { level: 1 })
            .innerHTML).toContain(title)
    });

    test('Debe de mostrar el subtítulo enviado por props', () => {
        render( <SecondApp 
                title={title}
                subtitle={subtitle} 
            />);
        expect( screen.getAllByText(subtitle).length).toBe(2);
    });

});