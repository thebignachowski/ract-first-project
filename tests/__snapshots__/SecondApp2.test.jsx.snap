// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SecondApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Nachowski
  </p>
</div>
`;
